syntax = "proto2";
import "nanopb.proto";

message SimplePODData {
    required uint32 value = 1 [(nanopb).int_size = IS_8];
}

message KindaComplexPODData {
    required uint32 value = 1 [(nanopb).int_size = IS_8];
    required uint32 value2 = 2 [(nanopb).int_size = IS_16];
    required uint32 value3 = 3 [(nanopb).int_size = IS_32];
    required uint64 value4 = 4;
}

message StringData {
    required string value = 1 [(nanopb).max_size = 32];
}

message GyroSensorData {
    required float x = 1;
    required float y = 2;
    required float z = 3;
}

message VariableSizedPODData {
    required uint32 value = 1 [(nanopb).int_size = IS_8];
    required bytes data = 2 [(nanopb).max_size = 32];
}
